core/templates/dev/head/pages/exploration_player/ImagePreloaderService.js

```diff
@@ -89,27 +124,29 @@ oppia.factory('ImagePreloaderService', [
       AssetsBackendApiService.loadImage(
         ExplorationContextService.getExplorationId(), imageFilename)
         .then(function(loadedImage) {
-          _filenamesOfImageCurrentlyDownloading = (
-            _filenamesOfImageCurrentlyDownloading.filter(
-              function(imageFilename) {
-                return loadedImage.filename !== imageFilename;
-              })
-          );
-          if (_filenamesOfImageToBeDownloaded.length > 0) {
-            var nextImageFilename = _filenamesOfImageToBeDownloaded.shift();
-            _filenamesOfImageCurrentlyDownloading.push(nextImageFilename);
-            _loadImage(nextImageFilename);
-          }
+          _removeCurrentAndLoadNextImage(loadedImage.filename);
           if (_imageLoadedCallback[loadedImage.filename]) {
             var onLoadImageResolve = (
               (_imageLoadedCallback[loadedImage.filename]).resolveMethod);
             var objectUrl = URL.createObjectURL(loadedImage.data);
             onLoadImageResolve(objectUrl);
             _imageLoadedCallback[loadedImage.filename] = null;
           }
+        }, function(filename) {
+          _failedDownload.push(filename);
+          _removeCurrentAndLoadNextImage(filename);
         });
     };
 
+    /**
+     * Checks if the given filename is in _failedDownload or not.
+     * @param {string} filename - The filename of the image which is to be
+     *                            removed from the _failedDownload array.
+     */
+    var _isInFailedDownload = function(filename) {
```

Comment:
Move this whole method before `_removeFromFailedDownload`

Reply:
Done!